### ISSUE TITLE
Does ELMo really boost reading comprehension performance ?

### ISSUE THREAD

**Issue by robbine** [allenai/allennlp#1842]
I tried to introduce elmo representation in bidaf by applying 
"elmo_representation = phrase_layer(highway_layer(elmo_embedder))"
 and append this elmo_representation along the last dimension to form a concatenated encoded_question and encoded_passage respectively.
Here is my code:
> 
		embedded_question = self._highway_layer(self._text_field_embedder(question))
		embedded_passage = self._highway_layer(self._text_field_embedder(passage))
		embedded_question_elmo = self._highway_layer_elmo(self._text_field_embedder_elmo(question))
		embedded_passage_elmo = self._highway_layer_elmo(self._text_field_embedder_elmo(passage))
		phrase_question = self._phrase_layer(embedded_question, question_lstm_mask)
		phrase_passage = self._phrase_layer(embedded_passage, passage_lstm_mask)
		phrase_question_elmo = self._phrase_layer_elmo(embedded_question_elmo, question_lstm_mask)
		phrase_passage_elmo = self._phrase_layer_elmo(embedded_passage_elmo, passage_lstm_mask)
		encoded_question = self._dropout(torch.cat((phrase_question, phrase_question_elmo), -1))
		encoded_passage = self._dropout(torch.cat((phrase_passage, phrase_passage_elmo), -1))

However it did not boost performance, Here are some metric curves, I read the paper which says there are two ways to adopt elmo, Is my way wrong ?
![image](https://user-images.githubusercontent.com/16010351/46253844-29da5080-c4b8-11e8-9efe-36ba5ac3c603.png)

**Comment by matt-peters** [comment]
Yes, ELMo improves SQuAD performance with bidaf > 4%.  See https://github.com/allenai/allennlp/blob/master/tutorials/how_to/elmo.md#reproducing-the-results-from-deep-contextualized-word-representations if you'd like to reproduce our published results.

**Comment by robbine** [comment]
I am running on a GPU, and it confuse me that a cudnn error occurs if I set requires_grad=true.
